src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -1279,10 +1279,16 @@ internal static void CalculateParentAssemblyDirectories(List<string> parentRefer
             // First, look for the dependency in the parents' directories. Unless they are resolved from the GAC or assemblyFoldersEx then
             // we should make sure we use the GAC and assemblyFolders resolvers themserves rather than a directory resolver to find the reference.
             // This way we dont get assemblies pulled from the GAC or AssemblyFolders but dont have the marking that they were pulled form there.
-            var parentReferenceFolders = new List<string>();
+            var parentReferenceDirectoriesMap = new Dictionary<string, List<string>>();
```

Comment:
I am not sure if the use of `Dictionary<string, List<string>>` is justified here. The dictionary groups directories by the same parent reference, but this is actually not really used anywhere. The only operation done on the dictionary is a "flattening" enumeration in `CompileDirectories`.

Could you consider something like `List<(string, string)>`, i.e. a list of tuples, instead? Note that duplicating the same parent full path in all dependencies costs us only one .NET reference (pointer size) whereas the overhead of a dictionary + inner lists is likely bigger. For CPU and code complexity it would be a win for sure.

Reply:
I agree. Thanks!